File: backend/app/Models/InventarioCategoria.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class InventarioCategoria extends Model
{
    use HasFactory;

    protected $table = 'inventario_categorias';

    protected $fillable = [
        'nombre',
        'descripcion',
        'activo',
    ];

    protected $casts = [
        'activo' => 'boolean',
    ];

    /**
     * Normaliza el nombre a UPPERCASE antes de guardar.
     */
    public function setNombreAttribute(string $value): void
    {
        $this->attributes['nombre'] = strtoupper(trim($value));
    }

    public function productos()
    {
        return $this->hasMany(InventarioProducto::class, 'categoria_id');
    }
}

File: frontend/lib/modules/inventario/models/inventario_categoria.dart
class InventarioCategoria {
  final int? id;
  final String nombre;
  final String? descripcion;
  final bool activo;

  const InventarioCategoria({
    this.id,
    required this.nombre,
    this.descripcion,
    this.activo = true,
  });

  factory InventarioCategoria.fromJson(Map<String, dynamic> json) {
    return InventarioCategoria(
      id: json['id'] as int?,
      nombre: (json['nombre'] as String?) ?? '',
      descripcion: json['descripcion'] as String?,
      activo: json['activo'] == true || json['activo'] == 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'nombre': nombre,
      if (descripcion != null) 'descripcion': descripcion,
      'activo': activo,
    };
  }
}
